subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/statistics/TaskExecutionStatisticsEventAdapter.java

```diff
@@ -49,10 +50,21 @@ public void beforeExecute(Task task) {
     public void afterExecute(Task task, TaskState state) {
         if (!taskIsForNestedBuild(task)) {
             TaskStateInternal stateInternal = (TaskStateInternal) state;
-            if (stateInternal.isAvoided()) {
-                avoidedTasksCount++;
-            } else if (stateInternal.isActionsWereExecuted()) {
-                executedTasksCount++;
+            if (stateInternal.isActionable()) {
+                switch (stateInternal.getOutcome()) {
```

Comment:
Before reverting to using the outcomes I tried to make it work with the methods on `TaskStateInternal`, but I ended up with brittle code (e.g. checking if the outcome was `getUpToDate()` must happen after checking if it was from cache) that was also hard to read. If you don't feel strongly, I'd leave it as it is. If there's a change in `TaskExecutionOutcome`, we'd need to update this code anyway (if only to add test cases to ensure that we report the right stuff), which we can do by searching for references for `TaskExecutionOutcome`. WDYT?